figures/pipeline/course_daily_metrics.py

```diff
@@ -344,11 +337,7 @@ def load(self, date_for=None, force_update=False, **_kwargs):
         Raises ValidationError if invalid data is attempted to be saved to the
         course daily metrics model instance
         """
-        if not date_for:
-            date_for = prev_day(
-                datetime.datetime.utcnow().replace(tzinfo=utc).date())
-        else:
-            date_for = as_datetime(date_for).replace(tzinfo=utc)
+        date_for = pipeline_date_for_rule(date_for)
```

Comment:
@OmarIthawi I pulled the common code out of course_daily_metrics and site_daily_metrics into a new module function, `figures.pipeline.helpers.pipeline_date_for_rule`. Yeah, so the name is kinda crappy, but I didn't want to spend a bunch of time on it. It at least isolates this logic rom the daily metrics loaders